File: training/src/classesandobjects/scala/inheritance/Customer.scala
package classesandobjects.scala.inheritance

class Customer(val name: String, val address: String) {
  private final val basket: Seq[String] = Seq()

  def add(item: String): Unit = {
    basket :+ item
  }

  def count: Double = {
    basket.size
  }
}

class DiscountedCustomer(name: String, address: String) extends Customer(name, address) {
  override def count: Double = super.count * 0.80
}

File: training/src/classesandobjects/java/Customer.java
package classesandobjects.java;

public class Customer {
    private final String name;
    private final String address;

    public Customer(String name, String address) {
        this.name = name;
        this.address = address;
    }

    public static void main(String[] args) {
        Customer bob = new Customer("Bob", "Alvear 1520");
    }
}
